mesheryctl/internal/cli/root/app/onboard.go

```diff
@@ -0,0 +1,288 @@
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/models"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	skipSave bool // skip saving a app
+	appFile  string
+)
+
+var onboardCmd = &cobra.Command{
+	Use:   "onboard",
+	Short: "Onboard application",
+	Long:  `Command will trigger deploy of Application file`,
+	Args:  cobra.MinimumNArgs(0),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var req *http.Request
+		var err error
+		client := &http.Client{}
+
+		// set default tokenpath for app onboard command.
+		if tokenPath == "" {
+			tokenPath = constants.GetCurrentAuthToken()
+		}
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return errors.Wrap(err, "error processing config")
+		}
+
+		deployURL := mctlCfg.GetBaseMesheryURL() + "/api/experimental/application/deploy"
+		appURL := mctlCfg.GetBaseMesheryURL() + "/api/experimental/application"
+
+		// app name has been passed
+		if len(args) > 0 {
+			// Merge args to get app-name
+			appName := strings.Join(args, "%20")
+
+			// search and fetch apps with app-name
+			log.Debug("Fetching apps")
+
+			req, err = http.NewRequest("GET", appURL+"?search="+appName, nil)
+			if err != nil {
+				return err
+			}
+
+			err = utils.AddAuthDetails(req, tokenPath)
+			if err != nil {
+				return errors.New("authentication token not found. please supply a valid user token with the --token (or -t) flag")
+			}
+
+			resp, err := client.Do(req)
+			if err != nil {
+				return err
+			}
+
+			var response *models.ApplicationsAPIResponse
+			// failsafe (bad api call)
+			if resp.StatusCode != 200 {
+				return errors.Errorf("Response Status Code %d, possible Server Error", resp.StatusCode)
+			}
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return errors.Wrap(err, utils.PerfError("failed to read response body"))
+			}
+			err = json.Unmarshal(body, &response)
+			if err != nil {
+				return errors.Wrap(err, "failed to unmarshal response body")
+			}
+
+			index := 0
+			if len(response.Applications) == 0 {
+				return errors.New("no apps found with the given name")
+			} else if len(response.Applications) == 1 {
+				appFile = response.Applications[0].ApplicationFile
+			} else {
+				// Multiple apps with same name
+				index = multipleApplicationsConfirmation(response.Applications)
+				appFile = response.Applications[index].ApplicationFile
+			}
+		} else {
+			// Method to check if the entered file is a URL or not
+			if validURL := govalidator.IsURL(file); !validURL {
+				content, err := ioutil.ReadFile(file)
+				if err != nil {
+					return err
+				}
+				text := string(content)
+
+				// if --skip-save is not passed we save the apps first
+				if !skipSave {
+					jsonValues, err := json.Marshal(map[string]interface{}{
+						"app_data": map[string]interface{}{
```

Comment:
The attribute is not "app_data" it is "application_data".

Reply:
updated attribute